test/unit/component.js

```diff
@@ -1,189 +1,212 @@
-      let node = document.createElement('div');
-      let component = new Component({
-        id: 1234,
-        handle: 'lol',
-        node: node,
-      });
-
-      assert.equal(component.id, 1234);
-      assert.equal(component.handle, 'lol');
-      assert.equal(component.node, node);
+    let component;
+    const config = {
+      id: 'id',
+      handle: 'handle',
+      storefrontId: 'sfid',
+      debug: 'debug',
+      cartNode: 'cartNode',
+      moneyFormat: '${{amount}}',
+      modalNode: 'modalNode',
+      toggles: 'toggles',
+      node: document.createElement('div'),
+      options: {
+        product: {
+          buttonDestination: 'modal',
+        },
+      },
+    };
+    const props = 'props';
+
+    beforeEach(() => {
+      component = new Component(config, props);
+    });
+
+    it('sets id, storeFrontId, node, and handle on instance', () => {
+      assert.equal(component.id, config.id);
+      assert.equal(component.handle, config.handle);
+      assert.equal(component.storefrontId, config.storefrontId);
+      assert.equal(component.node, config.node);
     });
 
     it('sets globalConfig based on passed in config', () => {
-      let component = new Component({
-        id: 1234
-      });
-      assert.equal(component.globalConfig.moneyFormat, '${{amount}}');
+      const expectedObj = {
+        debug: config.debug,
+        cartNode: config.cartNode,
+        moneyFormat: config.moneyFormat,
+        modalNode: config.modalNode,
+        toggles: config.toggles,
+      };
+      assert.deepEqual(component.globalConfig, expectedObj);
     });
 
     it('instantiates a view', () => {
-      let component = new Component({
-        id: 1234
-      });
       assert.instanceOf(component.view, View);
     });
 
-    it('merges configuration options with defaults', () => {
-      const config = {
-        options: {
-          product: {
-            buttonDestination: 'modal',
-          }
-        }
-      }
-      let component = new Component(config);
-      assert.equal(component.config.product.buttonDestination, 'modal', 'configuration options override defaults');
-      assert.equal(component.config.cart.iframe, true, 'defaults are merged into configuration');
+    it('instantiates an updater', () => {
+      assert.instanceOf(component.updater, Updater);
     });
-  });
 
-  describe('get name()', () => {
-    it('returns name based on handle or id', () => {
-      let component = new Component({id: 1234});
-      assert.equal(component.name, 'frame-undefined-1234', 'uses ID if ID is set');
-      component = new Component({handle: 'lol'});
-      assert.equal(component.name, 'frame-undefined-lol', 'uses handle if handle is set');
+    it('sets config from merging config.options with componentDefaults', () => {
+      assert.equal(component.config.product.buttonDestination, config.options.product.buttonDestination);
+      assert.deepEqual(component.config.cart, componentDefaults.cart);
```

Comment:
Is it possible to stub out the componentDefaults' default, and do a deepEquals on some mocked return?